cupy/random/_generator_api.pyx

```diff
@@ -320,8 +325,34 @@ class Generator:
             <numpy.random.generator.Generator.poisson>`
         """
         cdef ndarray y
+        cdef ndarray lam_arr
+
+        if not isinstance(lam, ndarray):
+            if type(lam) in (float, int):
+                lam_a = ndarray((), numpy.float64)
+                lam_a.fill(lam)
+                lam = lam_a
+            else:
+                if lam is None:
+                    raise TypeError('lam must be real number, not NoneType')
+                raise ValueError('lam is required to be a cupy.ndarray'
+                                 ' or a scalar')
```

Comment:
`lam < 0` raises `ValueError`.  Is there any other case with `ValueError`?

Reply:
I think no, also we don't check the value of lam to avoid synchronization.

Should we check it if it's a regular scalar value instead of a ndarray?

Reply:
```suggestion
                raise TypeError('lam is required to be a cupy.ndarray'
                                ' or a scalar')
```

BTW, I found a `ValueError` case, but I didn't get NumPy's intention.
```
>>> np.random.default_rng().poisson(1+0j)
TypeError: can't convert complex to float
>>> np.random.default_rng().poisson('inf')
TypeError: must be real number, not str
>>> np.random.default_rng().poisson('foo')
ValueError: could not convert string to float: 'foo'
>>> np.__version__
'1.20.2'
```

Reply:
oh! I didn't though of testing string, I just assumed that all primitive types would result in `TypeError`
Good catch!

Reply:
should we match this behavior too?